pkg/kubelet/prober/worker.go

```diff
@@ -219,7 +221,14 @@ func (w *worker) doProbe(ctx context.Context) (keepGoing bool) {
 		return false
 	}
 
-	c, ok := podutil.GetContainerStatus(status.ContainerStatuses, w.container.Name)
+	containerStatuses := status.ContainerStatuses
+	if utilfeature.DefaultFeatureGate.Enabled(features.SidecarContainers) {
+		containerStatuses = make([]v1.ContainerStatus, 0, len(status.ContainerStatuses)+len(status.InitContainerStatuses))
```

Comment:
Why is this changing?  Not clear to me (but it's late!).

Do ephemeral containers not need to be in this list because they cannot have probes?

Reply:
Init containers and ephemeral containers were originally not allowed to have probes.
As part of this enhancement, the restartable init container can have probes.